tests/python_package_test/test_arrow.py

```diff
@@ -432,3 +432,15 @@ def test_predict_ranking():
         num_boost_round=5,
     )
     assert_equal_predict_arrow_pandas(booster, data)
+
+
+def test_arrow_categorical():
+    data = generate_random_arrow_table(10, 10000, 42)
+    dataset = lgb.Dataset(
+        data,
+        label=generate_random_arrow_array(10000, 43, generate_nulls=False, values=np.arange(4)),
```

Comment:
```suggestion
        label=generate_random_arrow_array(num_datapoints=data.shape[0], seed=43, generate_nulls=False, values=np.arange(4)),
```

Let's please have this reference the size of the feature data, instead of repeating it.